sql/catalyst/src/test/java/org/apache/spark/sql/catalyst/expressions/RowBasedKeyValueBatchSuite.java

```diff
@@ -338,15 +338,18 @@ public void appendRowUntilExceedingCapacity() throws Exception {
 
   @Test
   public void appendRowUntilExceedingPageSize() throws Exception {
+    // Use of this property prevents issues when running on a two core machine
+    // A developer may override this size to 1 MB not 64 MB, so use that if so
+    int pageSizeToUse = Integer.parseInt(System.getProperty("spark.buffer.pageSize"));
```

Comment:
Running the tests now with 
```
int pageSizeToUse = (int) memoryManager.pageSizeBytes(); 
```
Looking at the logic in here I see we either use the default page size or whatever the user specifies

An int is accepted for the allocate method in RowBasedKeyValueBatch, given that a max int in Java is 2,147m I doubt this will be a problem